language/move-prover/bytecode-to-boogie/src/prelude.bpl

```diff
@@ -0,0 +1,543 @@
+// ================================================================================
+// Domains
+
+type TypeName;
+type FieldName = int;
+type LocalName;
+type Address = int;
+type ByteArray;
+type String;
+type {:datatype} Edge;
```

Comment:
What does `{:datatype}` say to the solver?

Reply:
It indicates that an inductive datatype is being declared. A collection of constructors have to be defined for each type.